File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        
        left_running_product = 1
        for i in range(len(nums)):
            res[i] = left_running_product
            left_running_product *= nums[i]

        right_running_product = 1
        for i in range(len(nums)-1, -1, -1):
            res[i] *= right_running_product
            right_running_product *= nums[i]

        return res
        
        

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> list = new HashSet(nums.length);

        for (int i = 0; i < nums.length; i++) {
            if (!list.add(nums[i])) {
                return true;
            }
        } 

        return false;
    }
}
